master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -954,9 +952,10 @@ private[celeborn] class Master(
     logInfo(s"Offer slots successfully for $numReducers reducers of $shuffleKey" +
       s" on ${slots.size()} workers.")
 
-    val workersNotSelected = availableWorkers.asScala.filter(!slots.containsKey(_))
-    val offerSlotsExtraSize = Math.min(conf.masterSlotAssignExtraSlots, workersNotSelected.size)
+    val workersNotSelectedSize = availableWorkers.size() - slots.size()
```

Comment:
Using the `workersNotSelected` here won't be a trouble as you need the unused worker list.

Reply:
done